src/components/mobile-menu/mobile-menu.jsx

```diff
@@ -0,0 +1,153 @@
+import React, { Component } from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import { Spring } from 'react-spring'
+import { Flex } from 'grid-styled'
+import { Portal } from 'react-portal'
+
+import SwipeEventEmitter from '../swipe.event'
+import { MenuIcon } from '../icons'
+
+let MobileSidebar = styled.div.attrs({
+  style: ({ translation }) => ({
+    transform: `translateX(${translation || 0}px)`,
+  }),
+})`
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  position: fixed;
+  top: 0;
+  height: 100vh;
+  width: 100%;
+  left: 100%;
+  overflow: auto;
+  background: #fff;
+
+  transition: ${props =>
+    props.transitioned ? 'all 50ms ease-in-out' : 'none'};
+`
+
+class MobileMenu extends Component {
+  state = {
+    isMenuOpen: false,
+    drawerPosition: 0,
+  }
+
+  static propTypes = {
+    stickyOffset: PropTypes.number,
+  }
+
+  static defaultProps = {
+    stickyOffset: 75,
+  }
+
+  componentDidMount() {
+    document.body.style.overflow = 'visible'
+  }
+
+  render() {
+    return (
+      <React.Fragment>
+        <SwipeEventEmitter
+          onRelease={this.onSwipeRelease}
+          onHorizontalMove={this.updateDrawerPosition}
+        />
+        <Flex justifyContent="space-between" alignItems="center">
+          {this.props.renderLogo()}
+          <MenuIcon onToggle={this.toggle} isOpened={this.state.isMenuOpen} />
+        </Flex>
+        <Portal>
+          <Spring
+            to={this.animationPose}
+            immediate={name =>
+              this.state.drawerPosition !== 0 && name === 'translation'
+            }>
+            {({ translation, opacity }) => (
+              <MobileSidebar
+                translation={translation}
+                transitioned={this.state.drawerPosition === 0}>
+                {this.isSticky ? (
```

Comment:
why do you use getters instead of functions?

Reply:
Looks cooler :) I don't know actually, find it interesting

Reply:
I think it is less clear, and should only be used only where it is needed